src/test/java/uk/gov/companieshouse/registeredemailaddressapi/unit/service/MockEmailRetrieverImplTest.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.companieshouse.registeredemailaddressapi.unit.service;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import uk.gov.companieshouse.api.model.company.RegisteredEmailAddressJson;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.ServiceException;
+import uk.gov.companieshouse.registeredemailaddressapi.service.MockEmailRetrieverImpl;
+
+class MockEmailRetrieverImplTest {
+
+    private static final String COMPANY_NUMBER = "12345678";
+    private static final String COMPANY_NUMBER_ERR = "12345ERR";
+    private static final String COMPANY_EMAIL_MOCK = "mockexistingemail@companieshouse.gov.uk";
+
+
+    private MockEmailRetrieverImpl mockEmailRetrieverImpl = new MockEmailRetrieverImpl();
+
+
+    @Nested
```

Comment:
cannot see the need for this, same comment for the Oracle impl test class

Reply:
The existing (renamed in this PR) Oracle impl test class had this nesting first, this parallel one is just following the same structure.  In general I like the clarity added to test output by this kind of grouping, but agree that it isn't needed here as neither has more than one group of tests.  I am inclined to leave it here as it isn't doing any harm, unless you feel strongly?

Reply:
As per your preference